src/lib/metrics.rs

```diff
@@ -370,12 +370,19 @@ impl<'a> DemuxedGroupMetrics<'a> {
             let filename =
                 [prefix.to_string(), "sample_barcode_hop_metrics.tsv".to_string()].concat();
             let output_path = output_dir.as_ref().join(filename);
+
+            let b1_len = sample_barcode_hop_tracker.checker.index1_length;
             delim.write_tsv(
                 &output_path,
                 sample_barcode_hop_tracker
                     .count_tracker
                     .into_iter()
-                    .map(|(barcode, count)| BarcodeCount::new(barcode.into(), count as isize))
+                    .map(|(barcode, count)| {
+                        let b1 = BString::from(&barcode[..b1_len]);
+                        let b2 = BString::from(&barcode[b1_len..]);
+                        let barcode = BString::from(format!("{}+{}", b1, b2));
```

Comment:
@nh13 @jiangweiyao, where hopped barcodes get delimited.

Reply:
1.  Now we have two places where this formatting occurs, once in the sample metadata and once in the metric modules.  Can you refactor to once place please and call from both?
2. What happens if `b2` is empty?  I don't think this is doing the right thing.

Reply:
1. I strongly agree that this is code-smell coming from different barcode representations. Finding a solution might take more time / effort.
2. For hop metrics, we know that the reported barcodes can be split and delimited. It is only used for dual index barcodes. The `SampleBarcodeHopTracker` only updates counts for delimited and hopped barcodes. See: https://github.com/Singular-Genomics/singular-demux/blob/e9706b6a25a361fef2b8a7f40d8c4b537e78dd72/src/lib/metrics.rs#L655